usr/sbin/p11sak/p11sak.c

```diff
@@ -4485,19 +4498,33 @@ static CK_RV get_obj_infos(CK_OBJECT_HANDLE obj, CK_OBJECT_CLASS *class,
     return CKR_OK;
 }
 
-
+#if defined(_AIX)
+static int iterate_compare(const void *a, const void *b)
+#else
 static int iterate_compare(const void *a, const void *b, void *private)
+#endif
 {
+#if defined(_AIX)
+    struct p11sak_iterate_compare_data *data = global_private;
+#else
     struct p11sak_iterate_compare_data *data = private;
+#endif
     const CK_OBJECT_HANDLE *obj1 = a;
     const CK_OBJECT_HANDLE *obj2 = b;
     int result = 0;
     CK_RV rc;
 
-    if (data->rc!= CKR_OK)
+    if (data->rc != CKR_OK)
         return 0;
 
+#if defined(_AIX)
+    void *oldptr = global_private;
```

Comment:
please move the definition of the oldptr variable above to the other variables (inside the #if defined(_AIX) block).